### PR TITLE
Frepp wrapper for MDTF

### PR DESCRIPTION
**Pull request by aradhakrishnanGFDL**
**Description**

GFDL MDTF FREPP wrapper script update, includes a gfdl mdtf frepp template jsonc.

Associated issue #349  

**How Has This Been Tested?**
Yes

**Checklist:**
- [ ] I have reviewed my own code to ensure that if follows the [POD development guidelines](https://mdtf-diagnostics.readthedocs.io/en/latest/sphinx/dev_guidelines.html)
- [ ] My branch is up-to-date with the NOAA-GFDL develop branch, and all merge conflicts are resolved
- [ ] The script are written in Python 3.6 or above (preferred; required if funded by a CPO grant), NCL, or R
- [ ] All of my scripts are in the diagnostics/[POD short name] subdirectory, and include a main_driver script, template html, and settings.jsonc file
- [ ] I have made corresponding changes to the documentation in the POD's doc/ subdirectory
- [ ] If applicable, I've added a .yml file to src/conda, and my environment builds with `conda_env_setup.sh` 
- [ ] I have added any necessary data to input_data/obs_data/[pod short name] and/or input_data/model/[pod short name]
- [ ] My code is portable; it uses [MDTF environment variables](https://mdtf-diagnostics.readthedocs.io/en/latest/sphinx/ref_envvars.html), and does not contain hard-coded file or directory paths
- [ ] I have provided the code to generate digested data files from raw data files
- [ ] Each digested data file generated by the script contains numerical data (no figures), and is 3 GB or less in size
- [ ] The repository contains no extra test scripts or data files

### CODE REVIEW

**Comment by aradhakrishnanGFDL** [comment]
testing notes in #349

File: sites/NOAA_GFDL/mdtf_gfdl.csh [hunk 1]
```diff
@@ -144,26 +157,47 @@ wipetmp
...
+sed -i 's/LASTYR1/'${yr2}'/g' $input_jsonc
+sed -i 's|OUTPUTDIR1|'${out_dir}'|' $input_jsonc
+sed -i 's/CONVENTION1/'${conv}'/g' $input_jsonc
+sed -i 's/POD_LIST/'${pod_list}'/g' $input_jsonc
+
+echo "Filled in input settings json and using this for the MDTF run $input_jsonc"
+
+gcp -cd $input_jsonc ${out_dir}/
+
+if (! -d $in_data_dir) then
```
Comment (ceblanton) [inline sites/NOAA_GFDL/mdtf_gfdl.csh:190]: missing `endif`

File: sites/NOAA_GFDL/mdtf_frepp_template.jsonc [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  // If running on a GFDL workstation, set $MDTF_TMPDIR to this location
+  // and create temp files here. Must be accessible via gcp.
+  //"GFDL_WS_TEMP": "$MDTF_TMPDIR",
+  "GFDL_WS_TEMP": "/archive/$USER/tmp",
+
+  // Set flag to run framework in 'online' mode, processing data as part of the
+  // FRE pipeline. Normally this is done by calling the mdtf_gfdl.csh wrapper
+  // script from the XML.
+  "frepp": false,
+
```
Comment (jkrasting) [inline sites/NOAA_GFDL/mdtf_frepp_template.jsonc:68]:
```suggestion

// Allow the framework to pull files/variables from multiple components
"any_components": true,

```

Comment (jkrasting) [reply sites/NOAA_GFDL/mdtf_frepp_template.jsonc:68]: I just came across this JSON option today.  In order for PODs to operate across components, this flag will need to be set.  Might as well set this now.

**Review by jkrasting — CHANGES_REQUESTED** [review]

File: sites/NOAA_GFDL/mdtf_gfdl.csh [hunk 3]
```diff
@@ -25,11 +25,24 @@ set dataendyr
 set datachunk
 set staticfile
 set fremodule
+set conv = GFDL
+set model = "modelname"
+
+# Please enter the POD lists below. This gets filled in to the MDTF input json automatically if using frepp
+#this is tested with one POD only at this time
+
+set pod_list = '"EOF_500hPa"'
```
[the remark this answers is not in the document]
Comment (aradhakrishnanGFDL) [reply sites/NOAA_GFDL/mdtf_gfdl.csh:34]: yes per @Wen-hao-Dong's email exchange yesterday

[the remark this answers is not in the document]
Comment (Wen-hao-Dong) [reply sites/NOAA_GFDL/mdtf_gfdl.csh:34]: Yes, that is correct. By setting set pod_list = '"EOF_500hPa", "precip_diurnal_cycle"', we get both PODs run successfully.

[the remark this answers is not in the document]
Comment (wrongkindofdoctor) [reply sites/NOAA_GFDL/mdtf_gfdl.csh:34]: @Wen-hao-Dong @aradhakrishnanGFDL Thank you!

**Review by wrongkindofdoctor — APPROVED** [review]

**Pull request merged**